### PR TITLE
Add User-Agent header to all requests

### PR DESCRIPTION
**Pull request by AndriyShevchuk**
* Add User-Agent to headers for all requests. User does not able to change it.
* Update unit-tests and integration tests

Resolves: OLPEDGE-1707
Signed-off-by: ashevchu <ext-andrii.shevchuk@here.com>

### CODE REVIEW

File: @here/olp-sdk-authentication/lib/UserAuth.ts [hunk 1]
```diff
@@ -241,7 +241,8 @@ export class UserAuth {
 
         const headers = new Headers({
             Authorization: "Bearer " + token,
-            "Content-Type": "application/json"
+            "Content-Type": "application/json",
+            "User-Agent": "OLP-TS-SDK/1.x.x"
```
Comment (andescu) [inline @here/olp-sdk-authentication/lib/UserAuth.ts]: The 1.x.x should actually be the version of the SDK, e.g. 1.3.0 or 1.3.1.
It definitely should not be "1.x.x".

File: @here/olp-sdk-authentication/app.version.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+// tslint:disable-next-line: no-var-requires
+export const APP_VERSION = require("../../package.json").version;
```
Comment (OleksiiZubko) [inline @here/olp-sdk-authentication/app.version.ts]: Please use string "1.4.0", do not require package.json

File: @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts [hunk 3]
```diff
@@ -50,6 +51,32 @@ async function addBearerToken(
...
 }
 
+/**
+ * Adds user-agent to the request headers
+ * @param init params to the request. @see RequestInit interface.
+ *
+ * @returns Promise with updated request params (with added user-agent to the headers).
+ */
+async function addUserAgent(init?: RequestInit): Promise<RequestInit> {
+    const USER_AGENT = `OLP-TS-SDK/${APP_VERSION}`;
```
[the remark this answers is not in the document]
Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts]: Function returns promise so we do need async.

File: @here/olp-sdk-authentication/app.version.ts (hunk 2 above)
Comment (AndriyShevchuk) [reply @here/olp-sdk-authentication/app.version.ts]: Done

File: @here/olp-sdk-authentication/lib/UserAuth.ts [hunk 4]
```diff
@@ -268,7 +271,8 @@ export class UserAuth {
     async getUserInfo(userToken: string): Promise<UserInfo> {
         const headers = new Headers({
             Authorization: "Bearer " + userToken,
-            "Content-Type": "application/json"
+            "Content-Type": "application/json",
+            "User-Agent": this.USER_AGENT
```
[the remark this answers is not in the document]
Comment (AndriyShevchuk) [reply @here/olp-sdk-authentication/lib/UserAuth.ts]: Done

File: @here/olp-sdk-authentication/lib/UserAuth.ts [hunk 5]
```diff
@@ -141,6 +142,7 @@ export class UserAuth {
     private m_credentials: AuthCredentials = {};
     private m_scope?: string;
     private readonly m_apiUrl: string;
+    private USER_AGENT = `OLP-TS-SDK/${APP_VERSION}`;
```
[the remark this answers is not in the document]
Comment (AndriyShevchuk) [reply @here/olp-sdk-authentication/lib/UserAuth.ts]: Done

File: @here/olp-sdk-authentication/lib/requestToken_common.ts [hunk 6]
```diff
@@ -158,7 +160,8 @@ export async function requestToken_common(
 
     const headers = new Headers({
         Authorization: authorization,
-        "Content-Type": "application/json"
+        "Content-Type": "application/json",
+        "User-Agent": USER_AGENT
```
[the remark this answers is not in the document]
Comment (AndriyShevchuk) [reply @here/olp-sdk-authentication/lib/requestToken_common.ts]: Done

File: @here/olp-sdk-dataservice-read/app.version.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+// tslint:disable-next-line: no-var-requires
+export const APP_VERSION = require("../../package.json").version;
```
[the remark this answers is not in the document]
Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/app.version.ts]: Done

File: @here/olp-sdk-authentication/lib/UserAuth.ts (hunk 1 above)
Comment (AndriyShevchuk) [reply @here/olp-sdk-authentication/lib/UserAuth.ts]: Updated

File: @here/olp-sdk-authentication/lib.version.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+// tslint:disable-next-line: no-var-requires
```
Comment (OleksiiZubko) [inline @here/olp-sdk-authentication/lib.version.ts]: we don't need than anymore

**Pull request merged**